File: strategy_pattern/murim/Duck.kt
abstract class Duck : QuackBehavior, FlyBehavior {

    fun Duck() {
    }

    abstract fun display()

    fun performFly() {
        fly()
    }

    fun performQuack() {
        quack()
    }

    fun swim() {
        println("오리는 물에 뜬다. 진짜 오리, 러버덕 둘 다..")
    }
}

File: strategy_pattern/sese/Duck.py
from abc import *

# 추상 클래스로 duck class 구현
class Duck(metaclass=ABCMeta):
    @abstractmethod
    def display(self):
        pass

    # 나는 행동 상속
    def performFly(self):
        self.fly_behavior.fly()
    
    # 우는 행동 상속
    def performQuack(self):
        self.quack_behavior.quack()

    def swim(self):
        print("모든 오리는 수영할 줄 압니다. 둥둥둥...")

